src/Test/Functional/Acceptance/SplitDbCest.php

```diff
@@ -0,0 +1,279 @@
+            'ERROR: Enabling a split database will be skipped.'
+            . ' Relationship do not have configuration for next types: sales, quote'
+        );
+        $this->checkEnvPhpConfig($I, [], ['checkout', 'sales']);
+        $this->checkMagentoFront($I);
+    }
+
+    /**
+     * @param CliTester $I
+     * @param array $envMagento
+     * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+     * @throws TaskException
+     */
+    private function partWithSplitDbArch(CliTester $I, array $envMagento)
+    {
+        $I->runEceDockerCommand('build:compose --mode=production');
+
+        // Run splitting database for 'quote' tables
+        $envMagento['stage']['deploy']['SPLIT_DB'] = ['quote'];
+        $I->writeEnvMagentoYaml($envMagento);
+        $I->startEnvironment();
+        $I->runDockerComposeCommand('run deploy cloud-deploy');
+        $I->seeInOutput([
+            'INFO: Quote tables were split to DB magento2 in db-quote',
+            'INFO: Running setup upgrade.',
+        ]);
+        $this->checkEnvPhpConfig($I, ['checkout'], ['sales']);
+        $this->checkMagentoFront($I);
+
+        $I->stopEnvironment(true);
+
+        // 'Split Db' type was deleted
+        unset($envMagento['stage']['deploy']);
+        $I->writeEnvMagentoYaml($envMagento);
+        $I->startEnvironment();
+        $I->runDockerComposeCommand('run deploy cloud-deploy');
+        $I->seeInOutput('WARNING: Variable SPLIT_DB does not have data which were already split types: quote');
+        $this->checkEnvPhpConfig($I, ['checkout'], ['sales']);
+        $this->checkMagentoFront($I);
+
+        $I->stopEnvironment(true);
+
+        // 'Split Db' current type was deleted and new type added
+        $envMagento['stage']['deploy']['SPLIT_DB'] = ['sales'];
+        $I->writeEnvMagentoYaml($envMagento);
+        $I->startEnvironment();
+        $I->runDockerComposeCommand('run deploy cloud-deploy');
+        $I->seeInOutput('WARNING: Variable SPLIT_DB does not have data which were already split types: quote');
+        $this->checkEnvPhpConfig($I, ['checkout'], ['sales']);
+        $this->checkMagentoFront($I);
+
+        $I->stopEnvironment(true);
+
+        // 'Split Db' current type was returned
+        $envMagento['stage']['deploy']['SPLIT_DB'] = ['sales', 'quote'];
+        $I->writeEnvMagentoYaml($envMagento);
+        $I->startEnvironment();
+        $I->runDockerComposeCommand('run deploy cloud-deploy');
+        $I->seeInOutput([
+            'INFO: Sales tables were split to DB magento2 in db-sales',
+            'INFO: Running setup upgrade.',
+        ]);
+        $this->checkEnvPhpConfig($I, ['checkout', 'sales']);
+        $this->checkMagentoFront($I);
+    }
+
+    /**
+     * @param CliTester $I
+     * @param array $envMagento
+     * @throws Exception
+     */
+    private function partInstallWithSplitDb(CliTester $I, array $envMagento): void
+    {
+        $envMagento['stage']['deploy']['SPLIT_DB'] = ['sales'];
```

Comment:
Same here, I think these could be split out and reused in partWithoutSplitDbArch

Reply:
Done